pkg/infrastructure/gcp/clusterapi/firewallrules.go

```diff
@@ -215,31 +215,42 @@ func deleteFirewallRule(ctx context.Context, name, projectID string) error {
 	return nil
 }
 
+func hasFirewallPermission(ctx context.Context, projectID string) (bool, error) {
+	client, err := gcpconfig.NewClient(context.Background())
```

Comment:
nit: there is a ctx passed into this function, so creating a new background context should not be necessary. The calling function should not be setting timeouts or limiting the ctx passed in so that it cannot be used to create the client.